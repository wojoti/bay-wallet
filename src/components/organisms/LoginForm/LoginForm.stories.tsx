import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import LoginForm from './LoginForm';

export default {
  title: 'Organisms/LoginForm',
  component: LoginForm,
  argTypes: {onLoginSubmit: {action: 'onLoginSubmit'}},
} as ComponentMeta<typeof LoginForm>;
const Template: ComponentStory<typeof LoginForm> = args => (
  <LoginForm {...args} />
);

export const Username = Template.bind({});
Username.args = {testId: ''};

export const Password = Template.bind({});
Password.args = {testId: ''};
