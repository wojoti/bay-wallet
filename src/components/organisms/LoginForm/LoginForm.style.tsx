import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  buttonMargin: {
    marginVertical: 15,
  },
});
export default style;
