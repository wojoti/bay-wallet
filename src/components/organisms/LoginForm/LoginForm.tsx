import {Button, Container, Header} from 'components/atoms';
import {InputArea} from 'components/molecules';
import {PropsWithChildren, useState} from 'react';
import style from './LoginForm.style';

export type LoginFormProps = PropsWithChildren<{
  onLoginSubmit: (username: string, password: string) => void;
  testId?: string;
}>;

const LoginForm = ({onLoginSubmit, testId}: LoginFormProps) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const onSubmit = () => {
    onLoginSubmit(email, password);
  };
  const onEmailChange = (value: string) => {
    setEmail(value);
  };
  const onPasswordChange = (value: string) => {
    setPassword(value);
  };
  return (
    <Container testId={testId}>
      <Header testId="loginform-header">Login</Header>
      <InputArea
        type="username"
        onChange={onEmailChange}
        testId="loginform-inputarea-username"
      />
      <InputArea
        type="password"
        onChange={onPasswordChange}
        testId="loginform-inputarea-password"
      />
      <Container customStyle={style.buttonMargin}>
        <Button onPress={onSubmit} type={'primary'} testId="loginform-button">
          SIGN IN
        </Button>
      </Container>
    </Container>
  );
};

export default LoginForm;
