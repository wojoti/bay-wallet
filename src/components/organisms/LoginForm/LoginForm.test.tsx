import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import LoginForm, {LoginFormProps} from './LoginForm';

const mockOnLoginSubmit = jest.fn<
  LoginFormProps['onLoginSubmit'],
  [email: string, password: string]
>();

const props: LoginFormProps = {
  onLoginSubmit: mockOnLoginSubmit,
  testId: 'test-loginform-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<LoginForm {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render loginform', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
});

test('should render loginform - header', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const loginformHeaderElement =
    within(loginformElement).getByTestId('loginform-header');
  expect(loginformHeaderElement).toBeOnTheScreen();
});

test('should render loginform - button', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const loginformButtonElement =
    within(loginformElement).getByTestId('loginform-button');
  expect(loginformButtonElement).toBeOnTheScreen();
});

test('should render loginform - inputarea for username - label', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const inputareaElement = within(loginformElement).getByTestId(
    'loginform-inputarea-username',
  );
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaLabelElement =
    within(inputareaElement).getByTestId('inputarea-label');
  expect(inputareaLabelElement).toBeOnTheScreen();
});

test('should render loginform - inputarea for username - input', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const inputareaElement = within(loginformElement).getByTestId(
    'loginform-inputarea-username',
  );
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaInputElement =
    within(inputareaElement).getByTestId('inputarea-input');
  expect(inputareaInputElement).toBeOnTheScreen();
});

test('should render loginform - inputarea for password - label', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const inputareaElement = within(loginformElement).getByTestId(
    'loginform-inputarea-password',
  );
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaLabelElement =
    within(inputareaElement).getByTestId('inputarea-label');
  expect(inputareaLabelElement).toBeOnTheScreen();
});

test('should render loginform - inputarea for password - input', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const inputareaElement = within(loginformElement).getByTestId(
    'loginform-inputarea-password',
  );
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaInputElement =
    within(inputareaElement).getByTestId('inputarea-input');
  expect(inputareaInputElement).toBeOnTheScreen();
});

test('should render loginform - button - handle onPress', () => {
  render(<LoginForm {...props} />);
  const loginformElement = screen.getByTestId('test-loginform-id');
  expect(loginformElement).toBeOnTheScreen();
  const inputareaElement = within(loginformElement).getByTestId(
    'loginform-inputarea-username',
  );
  expect(inputareaElement).toBeOnTheScreen();
  const inputareaInputElement =
    within(inputareaElement).getByTestId('inputarea-input');
  expect(inputareaInputElement).toBeOnTheScreen();
  fireEvent.changeText(inputareaInputElement, 'randomlogin');

  const inputareaPasswordElement = within(loginformElement).getByTestId(
    'loginform-inputarea-password',
  );
  expect(inputareaPasswordElement).toBeOnTheScreen();
  const inputareaPasswordInputElement = within(
    inputareaPasswordElement,
  ).getByTestId('inputarea-input');
  expect(inputareaPasswordInputElement).toBeOnTheScreen();
  fireEvent.changeText(inputareaPasswordInputElement, 'random_password');

  const loginformButtonElement =
    within(loginformElement).getByTestId('loginform-button');
  expect(loginformButtonElement).toBeOnTheScreen();
  fireEvent.press(loginformButtonElement);
  expect(mockOnLoginSubmit).toHaveBeenCalledWith(
    'randomlogin',
    'random_password',
  );
});
