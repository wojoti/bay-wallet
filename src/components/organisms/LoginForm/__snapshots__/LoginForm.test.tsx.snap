// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-loginform-id"
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "Poppins-Bold",
        "fontSize": 24,
        "margin": 0,
        "padding": 0,
        "textAlign": "left",
      }
    }
    testID="loginform-header"
  >
    Login
  </Text>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="loginform-inputarea-username"
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "backgroundColor": "white",
            "flex": 0,
            "marginHorizontal": 15,
            "paddingHorizontal": 10,
            "top": 15,
            "width": "auto",
            "zIndex": 1,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Poppins-Light",
            "fontSize": 14,
            "margin": 0,
            "padding": 0,
            "textAlign": "left",
          }
        }
        testID="inputarea-label"
      >
        Username
      </Text>
    </View>
    <TextInput
      accessibilityHint="Allow to insert username"
      accessibilityLabel="Login input field"
      autoComplete="username"
      autoCorrect={false}
      onChangeText={[Function]}
      style={
        {
          "borderColor": "#363062",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontFamily": "Poppins-Light",
          "height": 60,
          "paddingLeft": 30,
          "width": "100%",
        }
      }
      testID="inputarea-input"
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="loginform-inputarea-password"
  >
    <Text
      style={
        {
          "color": "black",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "margin": 0,
          "padding": 0,
          "textAlign": "left",
        }
      }
      testID="inputarea-label"
    >
      Password
    </Text>
    <TextInput
      accessibilityHint="Allow to insert password"
      accessibilityLabel="Password input field"
      autoComplete="password"
      autoCorrect={false}
      onChangeText={[Function]}
      secureTextEntry={true}
      style={
        {
          "borderColor": "#363062",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontFamily": "Poppins-Light",
          "height": 60,
          "paddingLeft": 30,
          "width": "100%",
        }
      }
      testID="inputarea-input"
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#363062",
        "borderColor": "#363062",
        "borderRadius": 8,
        "borderStyle": "solid",
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="loginform-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "Poppins-Bold",
          "fontSize": 18,
          "margin": 0,
          "padding": 0,
          "textAlign": "center",
        }
      }
      testID="button-text"
    >
      SIGN IN
    </Text>
  </View>
</View>
`;
